R/tags.R

```diff
@@ -292,24 +327,35 @@ tagGetAttribute <- function(tag, attr) {
 #' @rdname tag
 #' @export
 tagAppendChild <- function(tag, child) {
+  tag <- assert_shiny_tag(tag)
   tag$children[[length(tag$children)+1]] <- child
   tag
 }
 
 #' @rdname tag
 #' @export
 tagAppendChildren <- function(tag, ..., list = NULL) {
+  tag <- assert_shiny_tag(tag)
   tag$children <- unname(c(tag$children, c(dots_list(...), list)))
   tag
 }
 
 #' @rdname tag
 #' @export
 tagSetChildren <- function(tag, ..., list = NULL) {
+  tag <- assert_shiny_tag(tag)
   tag$children <- unname(c(dots_list(...), list))
   tag
 }
 
+assert_shiny_tag <- function(tag) {
+  tag <- as.tags(tag)
```

Comment:
A few issues/questions:

* Why this line is necessary? It would help to have a comment explaining why this is here.
* If the function is going to modify the object and return it, a name with `assert` is a bit misleading, since assertions usually used to throw errors. Maybe `as_shiny_tag`?
* This function will also change a string into a tagList containing the string. From prior discussion, I think we were trying to avoid that.
* I guess my broader question is, what is the purpose of this function?